src/UI/Palette.elm

```diff
@@ -26,14 +57,25 @@ type Brightness
 
 color : Tone -> Brightness -> Color
 color tone brightness =
-    ( tone, brightness )
+    Color (Properties tone brightness) defaultOptions
 
 
 toElColor : Color -> Element.Color
-toElColor ( tone, brightness ) =
+toElColor (Color { tone, brightness } { alpha, contrast }) =
     tone
-        |> toColors
+        |> ifThenElse contrast contrastColors toColors
         |> getLum brightness
+        |> colorWithOpacity alpha
```

Comment:
I didn't realize this before, but I'm not sure if this is a good function name, as every color has opacity, so this is more an `updateColorOpacity`, `setColorOpacity` or something like that.